torch/csrc/api/include/torch/nn/options/activation.h

```diff
@@ -315,5 +315,96 @@ struct TORCH_API GumbelSoftmaxFuncOptions {
 
 } // namespace functional
 
+// ============================================================================
+
+/// Options for MultiheadAttention functional and module.
+struct TORCH_API MultiheadAttentionOptions {
+  MultiheadAttentionOptions(int64_t embed_dim, int64_t num_heads)
+    : embed_dim_(embed_dim), num_heads_(num_heads),
+      kdim_(embed_dim), vdim_(embed_dim) {}
+
+  /// total dimension of the model.
+  TORCH_ARG(int64_t, embed_dim);
+
+  /// parallel attention heads.
+  TORCH_ARG(int64_t, num_heads);
+
+  /// a Dropout layer on attn_output_weights. Default: 0.0.
+  TORCH_ARG(double, dropout) = 0.0;
+
+  /// add bias as module parameter. Default: true.
+  TORCH_ARG(bool, bias) = true;
+
+  /// add bias to the key and value sequences at dim=0.
+  TORCH_ARG(bool, add_bias_kv) = false;
+
+  /// add a new batch of zeros to the key and value sequences at dim=1.
+  TORCH_ARG(bool, add_zero_attn) = false;
+
+  /// total number of features in key. Default: c10::nullopt.
+  TORCH_ARG(int64_t, kdim);
```

Comment:
where is it declared as an optional variable? I am thinking if we should change it to:
``TORCH_ARG(c10::optional<int64_t>, kdim) = c10::nullopt;``

Reply:
https://github.com/pytorch/pytorch/pull/27309#discussion_r356274944